### ISSUE TITLE
Why does "ping" not work?

### ISSUE THREAD

**Issue by soarteam** [maruos/maruos#60]
ping 8.8.8.8
100% packet loss
The network is normal.
apt-get can install pack

**Comment by pdsouza** [comment]
Thank you for your report! How did you install ping?

I am running ping just fine on v0.3:

```
maru@jessie:~$ sudo apt-get install iputils-ping
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following extra packages will be installed:
  libgnutls-deb0-28 libgnutls-openssl27
Suggested packages:
  gnutls-bin
The following NEW packages will be installed:
  iputils-ping libgnutls-openssl27
The following packages will be upgraded:
  libgnutls-deb0-28
1 upgraded, 2 newly installed, 0 to remove and 91 not upgraded.
Need to get 827 kB of archives.
After this operation, 308 kB of additional disk space will be used.
Do you want to continue? [Y/n] Y
Get:1 http://httpredir.debian.org/debian/ jessie/main libgnutls-deb0-28 armhf 3.3.8-6+deb8u4 [633 kB]
Get:2 http://httpredir.debian.org/debian/ jessie/main libgnutls-openssl27 armhf 3.3.8-6+deb8u4 [140 kB]
Get:3 http://httpredir.debian.org/debian/ jessie/main iputils-ping armhf 3:20121221-5+b2 [53.7 kB]
Fetched 827 kB in 0s (944 kB/s)    
debconf: delaying package configuration, since apt-utils is not installed
(Reading database ... 75454 files and directories currently installed.)
Preparing to unpack .../libgnutls-deb0-28_3.3.8-6+deb8u4_armhf.deb ...
Unpacking libgnutls-deb0-28:armhf (3.3.8-6+deb8u4) over (3.3.8-6+deb8u3) ...
Selecting previously unselected package libgnutls-openssl27:armhf.
Preparing to unpack .../libgnutls-openssl27_3.3.8-6+deb8u4_armhf.deb ...
Unpacking libgnutls-openssl27:armhf (3.3.8-6+deb8u4) ...
Selecting previously unselected package iputils-ping.
Preparing to unpack .../iputils-ping_3%3a20121221-5+b2_armhf.deb ...
Unpacking iputils-ping (3:20121221-5+b2) ...
Processing triggers for man-db (2.7.0.2-5) ...
Setting up libgnutls-deb0-28:armhf (3.3.8-6+deb8u4) ...
Setting up libgnutls-openssl27:armhf (3.3.8-6+deb8u4) ...
Setting up iputils-ping (3:20121221-5+b2) ...
Setcap worked! Ping(6) is not suid!
Processing triggers for libc-bin (2.19-18+deb8u6) ...

maru@jessie:~$ ping 8.8.8.8
PING 8.8.8.8 (8.8.8.8) 56(84) bytes of data.
64 bytes from 8.8.8.8: icmp_seq=1 ttl=47 time=28.2 ms
64 bytes from 8.8.8.8: icmp_seq=2 ttl=47 time=33.1 ms
64 bytes from 8.8.8.8: icmp_seq=3 ttl=47 time=33.0 ms
64 bytes from 8.8.8.8: icmp_seq=4 ttl=47 time=29.8 ms
^C
--- 8.8.8.8 ping statistics ---
4 packets transmitted, 4 received, 0% packet loss, time 3005ms
rtt min/avg/max/mdev = 28.250/31.066/33.101/2.090 ms
```

Any other info you can provide on your setup would be helpful!

**Comment by soarteam** [comment]
root@jessie:/etc/postgresql# apt-get install iputils-ping
Reading package lists... Done
Building dependency tree
Reading state information... Done
The following NEW packages will be installed:
  iputils-ping
0 upgraded, 1 newly installed, 0 to remove and 465 not upgraded.
Need to get 56.8 kB of archives.
After this operation, 100 kB of additional disk space will be used.
WARNING: The following packages cannot be authenticated!
  iputils-ping
Install these packages without verification? [y/N] y
Get:1 http://http.kali.org/kali/ kali-rolling/main iputils-ping armhf 3:20161105-1 [56.8 kB]
Fetched 56.8 kB in 16s (3546 B/s)
Unescaped left brace in regex is deprecated, passed through in regex; marked by <-- HERE in m/^(.*?)(\\)?\${ <-- HERE ([^{}]+)}(.*)$/ at /usr/share/perl5/Debconf/Question.pm line 72.
Unescaped left brace in regex is deprecated, passed through in regex; marked by <-- HERE in m/\${ <-- HERE ([^}]+)}/ at /usr/share/perl5/Debconf/Config.pm line 30.
debconf: delaying package configuration, since apt-utils is not installed
Selecting previously unselected package iputils-ping.
(Reading database ... 78192 files and directories currently installed.)
Preparing to unpack .../iputils-ping_3%3a20161105-1_armhf.deb ...
Unpacking iputils-ping (3:20161105-1) ...
Processing triggers for man-db (2.7.0.2-5) ...
Setting up iputils-ping (3:20161105-1) ...
root@jessie:/etc/postgresql# ping 192.168.0.1
socket: Permission denied
root@jessie:/etc/postgresql# ping 8.8.8.8
socket: Permission denied
root@jessie:/etc/postgresql#

**Comment by soarteam** [comment]
We confirmed that "Systemd" does not work.
Switch to the "upstart" because "systemd" error and the installation fails, eventually leading to not boot.

**Comment by soarteam** [comment]
Now the question is confirmed by "PostgreSQL".

![-7545be671ce56adc](https://cloud.githubusercontent.com/assets/11802769/22853898/3b5a7658-f027-11e6-9f2a-fe889856ffe5.jpg)

**Comment by soarteam** [comment]
Since Maru supports "Debian Fedora Gentoo Oracle Plamo Ubuntu".
Why not publish these systems acquiescence to open SSH LXC image?
Mirror the website of the system are not open SSH.

**Comment by soarteam** [comment]
I let Maru boot Ubuntu14.04, but not SSH.

**Comment by soarteam** [comment]
I don't have HDMI

**Comment by soarteam** [comment]
Debian Bug, systemd in the ARM can not function properly.

**Comment by soarteam** [comment]
Developed anywhere. No desktop . Minimal version.
Nice~

**Comment by soarteam** [comment]
LXC supports Systemd only from 1.1.

**Comment by soarteam** [comment]
What is Maru LXC Version

**Comment by pdsouza** [comment]
I think I see what's going on. The root user has not been added to the necessary groups to access raw sockets. If you try this as the default "maru" user, ping should work.

Please add the root account to `inet` and `inet_raw` groups:

    root# adduser root inet
    root# adduser root inet_raw

That should fix your ping issues. I will keep this issue open so that I add the root account to these groups by default, thanks for bringing it up.

Regarding LXC, v0.3 uses LXC 1.0.7.

**Comment by soarteam** [comment]
root@jessie:/home/maru# adduser root inet
Adding user `root' to group `inet' ...
Adding user root to group inet
Done.
root@jessie:/home/maru# adduser root inet_raw
The user `root' is already a member of `inet_raw'.
root@jessie:/home/maru# ping 8.8.8.8
socket: Permission denied
root@jessie:/home/maru#

**Comment by soarteam** [comment]
Systemd does not work with 1.0.7.
This can cause serious problems.
Many services do not start properly.
Redis PG. ...etc

**Comment by soarteam** [comment]
You can try to upgrade LXC.
Or use Ubuntu 14.04
To be honest, Ubuntu compatibility better than Debian.
 Themes：
![1486858775074](https://cloud.githubusercontent.com/assets/11802769/22858552/b74ceb1a-f086-11e6-92ea-e004adbe1d18.jpeg)
flatabulous-theme ultra-flat-icons

**Comment by soarteam** [comment]
maru@jessie:~$ ping 8.8.8.8
PING 8.8.8.8 (8.8.8.8) 56(84) bytes of data.
64 bytes from 8.8.8.8: icmp_seq=1 ttl=45 time=774 ms
64 bytes from 8.8.8.8: icmp_seq=2 ttl=45 time=784 ms
64 bytes from 8.8.8.8: icmp_seq=3 ttl=45 time=704 ms
64 bytes from 8.8.8.8: icmp_seq=4 ttl=45 time=775 ms
64 bytes from 8.8.8.8: icmp_seq=5 ttl=45 time=725 ms
64 bytes from 8.8.8.8: icmp_seq=6 ttl=45 time=545 ms
64 bytes from 8.8.8.8: icmp_seq=7 ttl=45 time=365 ms
64 bytes from 8.8.8.8: icmp_seq=8 ttl=45 time=724 ms
64 bytes from 8.8.8.8: icmp_seq=9 ttl=45 time=615 ms
64 bytes from 8.8.8.8: icmp_seq=10 ttl=45 time=546 ms
^C
--- 8.8.8.8 ping statistics ---
10 packets transmitted, 10 received, 0% packet loss, time 8998ms
rtt min/avg/max/mdev = 365.090/656.175/784.465/129.270 ms
maru@jessie:~$

**Comment by pdsouza** [comment]
You need to reboot after adding root to those groups for them to take effect!

**Comment by pdsouza** [comment]
Nice setup with Ubuntu!

How did you create the new desktop image? `lxc-create`? Do you have HDMI working? I am assuming you installed the maru debpkg in the ubuntu rootfs?

Please check out https://github.com/maruos/blueprints if you would like to add ubuntu as a compatible distro. I am open to pull requests on this!

**Comment by soarteam** [comment]
Download "LXC Ubuntu14.04 ARMHF"

https://mirrors.tuna.tsinghua.edu.cn/lxc-images/images/
Then replace the "/data/maru/..." file

https://mirrors.tuna.tsinghua.edu.cn/lxc-images/images/ubuntu/trusty/armhf/default/20170207_03%3A49/rootfs.tar.xz

**Comment by soarteam** [comment]
Use chroot to enter the Ubuntu
install OpenSSH
Or Start LXC on your Linux computer.
install OpenSSH

**Comment by soarteam** [comment]
I don't have HDMI. So I used chroot~

**Comment by soarteam** [comment]
New Bug.
After the normal Debian reboot, other devices will not be able to connect to Debian via SSH.
SSH needs to be connected locally. Other devices can be connected.

**Comment by soarteam** [comment]
You need The kernel ：github.com/soarteam/bricked-hammerhead

**Comment by soarteam** [comment]
And https://github.com/soarteam/ElementalX-N5
You kernel version is too low
Can't run LXC1.1 or higher and systemd

**Comment by pdsouza** [comment]
I was able to get postgresql running fine on Debian, the issue has nothing to do with systemd. It turns out that postgresql creates a new user `postgres` and then tries to create sockets as part of the server process...but it has not been added to the `inet` and `inet_raw` groups required for socket access on Maru OS. Please add this user to the needed groups:

```
maru@jessie:~$ sudo adduser postgres inet
Adding user `postgres' to group `inet' ...
Adding user postgres to group inet
Done.
maru@jessie:~$ sudo adduser postgres inet_raw
Adding user `postgres' to group `inet_raw' ...
Adding user postgres to group inet_raw
Done.
```

Then you can start postgresql server:

```
maru@jessie:~$ systemctl status postgresql
�� postgresql.service - PostgreSQL RDBMS
   Loaded: loaded (/lib/systemd/system/postgresql.service; enabled)
   Active: active (exited) since Sat 2017-02-18 16:02:27 UTC; 9h ago
 Main PID: 6817 (code=exited, status=0/SUCCESS)
   CGroup: /system.slice/postgresql.service
maru@jessie:~$ ps -efH | grep postgres
maru      8531   391  0 01:11 pts/1    00:00:00       grep postgres
postgres  7166     1  0 Feb18 ?        00:00:02   /usr/lib/postgresql/9.4/bin/postgres -D /var/lib/postgresql/9.4/main -c config_file=/etc/postgresql/9.4/main/postgresql.conf
postgres  7168  7166  0 Feb18 ?        00:00:00     postgres: checkpointer process                                                                                              
postgres  7169  7166  0 Feb18 ?        00:00:01     postgres: writer process                                                                                                    
postgres  7170  7166  0 Feb18 ?        00:00:01     postgres: wal writer process                                                                                                
postgres  7171  7166  0 Feb18 ?        00:00:01     postgres: autovacuum launcher process                                                                                       
postgres  7172  7166  0 Feb18 ?        00:00:02     postgres: stats collector process                                                                                           
```

I will look into finding a way so that any new users are automatically added to the `inet` and `inet_raw` groups for network permissions.

The kernel you linked to is also v3.4, the same as our kernel.

**Comment by pdsouza** [comment]
root is now added by default to `inet` and `inet_raw` groups: https://github.com/maruos/blueprints/commit/3054caa8918af1267c4961314d5d91856207f21e

**Comment by pdsouza** [comment]
New non-system users created with `adduser` are now automatically added to `inet` and `inet_raw` groups for networking access, as well as `sdcard_rw` for access to the shared `/mnt/sdcard`.

**Comment by soarteam** [comment]
iodine: open_tun: /dev/net/tun: No such file or directory: No such file or directory

**Comment by pdsouza** [comment]
You may need to create the tun device node yourself since it looks like LXC/systemd does not create that by default. You may also need to add a device allow rule in the container config for that device node number. Please feel free to open a separate issue for this and we can look at enabling /dev/net/tun by default.

**Comment by pdsouza** [comment]
Fixed in v0.4, please upgrade!
